Objects/genobject.c

```diff
@@ -740,7 +745,9 @@ gen_set_qualname(PyObject *self, PyObject *value, void *Py_UNUSED(ignored))
                         "__qualname__ must be set to a string object");
         return -1;
     }
-    Py_XSETREF(op->gi_qualname, Py_NewRef(value));
+    Py_BEGIN_CRITICAL_SECTION(self);
```

Comment:
Would you mind adding a comment explaining why XDecRefDelayed() is needed here? The qualname should be a simple Python str object, no?

Reply:
I'm not proposing anything, I don't understand well the purpose of XDecRefDelayed(). That's why I'm asking for a comment :-)